src/utils.ts

```diff
@@ -120,9 +155,44 @@ export function formatBytes(bytes: number) {
   return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
 }
 
-export interface FrontMatterResult {
-  frontMatter: string | null;
-  body: string | null;
+export function getLocalizedValue(doc: Document, item: any, key: string) {
+  return doc.locale ? item[`${key}@${doc.locale.id}`] || item[key] : item[key];
+}
+
+export function interpolate(pod: Pod, string: string, params: any) {
+  // Cache created functions to avoid memory leak.
+  const names = Object.keys(params);
+  const vals = Object.values(params);
+  const key = `${string}:${names}`;
+  if (pod.cache.interpolations[key]) {
+    return pod.cache.interpolations[key](...vals);
+  }
+  const func = new Function(...names, `return \`${string}\`;`);
+  pod.cache.interpolations[key] = func;
+  return func(...vals);
+}
+
+export function localizeData(data: any, locale: Locale) {
+  if (!data) {
+    return data;
+  }
+
+  if (DataType.isObject(data)) {
+    const newData: any = {};
+    for (const key of Object.keys(data)) {
+      newData[key] = localizeData(data[key], locale);
+    }
+    return newData;
+  } else if (DataType.isArray(data)) {
+    const newData = [...data];
+    for (const i in newData) {
```

Comment:
Is this line correct? I was under the assumtion that `for (const i in myArray)` would return the array indexes.

I just tried this in the console:

```
a = [1, 2, 3]
for (const i in a) { console.log(i) }
```

and got:

```
0
1
2
```

Reply:
It is correct for how I was doing the array since the `newData = [...data]` creates a copy of the array so I do not overwrite the original array, then I was going through each index since I needed to be able to put the changed value in the same spot of the array.

I've changed it now though so that I create an empty array and push each processed item of the original array into the new array. It may be more efficient as well.